site/app/models/Gradeable.php

```diff
@@ -1138,4 +1138,64 @@ public function getComponentsGradedBy($grader_id) {
         }
         return $return;
     }
+
+    /**
+     * Get an associative array of all the data needed to render this gradeable and its various components and their marks.
+     * Does not contain submission-specific data like comments or which marks are selected.
+     * @return array
+     */
+    public function getStaticData() {
```

Comment:
Do we need this function?  I'm pretty sure the `AbstractModel` class has a `ToArray()` method for this purpose.

Reply:
if we don't mind sharing the static vs grading data, that is.

Reply:
He's filtering out components with a -1 order (not sure when that would happen though).

Reply:
Was not aware of `toArray` but looking at it now it's a lot more data than we need or would want, especially for peer grading (if that ever happens).